src/main/java/org/opentripplanner/apis/gtfs/datafetchers/QueryTypeImpl.java

```diff
@@ -445,6 +447,15 @@ public DataFetcher<Object> node() {
             // TODO: Add geometry
             return new NearbyStop(stop, Integer.parseInt(parts[0]), null, null);
           }
+        case "Leg":
+          if (id.equals("null") || id.isBlank()) {
```

Comment:
Do we really want to treat the string "null" specially? In what case can this happen?

Reply:
It seems that Leg base class retuns null in getLegReference() method, and that value gets encoded into relay id. Decoding the relay id  produces a string value 'null'. 

I added the check to prevent unhandled exception when somebody tries to refetch non-transit legs. Maybe there is a more elegant way to handle such situations.

Reply:
Now the response for non-transit legs  is 
"data": {  "node": null }

Reply:
How about checking if it's a transit leg in `LegImpl` and never return the string "null" but the `null` value?

Reply:
Seems like a better solution. We should make the id field nullable in that case.

Reply:
I tried making id of Node type nullable and everything seems to work fine. 

Another option would be to encode null as an empty string in  LegReferenceSerializer, in which case id.isBlank() test would work. This might also fix problems in transmodel api.